### PR TITLE
Added feature to publish benchmark data to bigquery table.

### PR DESCRIPTION
**Pull request by utkarsharma2**
Closes: #434 

Earlier we were storing benchmark data in `/tmp/<some-file>.ndjson` file locally, but now we need to perform the benchmarks in the CI pipeline and hence we need the data to be persisted. With this PR we are publishing the benchmark data in the Bigquery table - `astronomer-dag-authoring.Benchmark.load_files_to_database`

Publishing is not enabled by default, because developers can run benchmarks locally but this data may not be intended to be published. In order to allow the publishing of data, an environment variable needs to be created `ASTRO_PUBLISH_BENCHMARK_DATA`.

Note - We assume the existence of a table in the Bigquery, and the schema of the table is tightly coupled with profiling data to be stored. If we decide to change the profiling data in the future, we may need to alter the table schema.

### CODE REVIEW

File: tests/benchmark/run.py [hunk 1]
```diff
@@ -25,6 +28,80 @@ def subtract(after_dict, before_dict):
     return {key: after_dict[key] - before_dict.get(key, 0) for key in after_dict}
 
 
+def prep_insert_statement(
+    df: pd.DataFrame, table_name="load_files_to_database", schema="Benchmark"
```
Comment (sunank200) [inline tests/benchmark/run.py:32]: What would happen when the user runs the same benchmarking script multiple times? `dag_id` would be same in that case as the row inserted before

**Review by sunank200 — COMMENTED** [review]
Overall LGTM.

I had a question - Reason to put the benchmark metrics to Big Query is to make it queryable?

**Comment by utkarsharma2** [comment]
@sunank200 - yes, the metrics are more useful to us if we make them queriable, we can later decide to do a weekly comparison on them and alert if they are less than a certain threshold.

File: tests/benchmark/run.py (hunk 1 above)
Comment (utkarsharma2) [reply tests/benchmark/run.py:32]: @sunank200 - I have logged an [issue](https://github.com/astronomer/astro-sdk/issues/460) for this bug and a few other bugs with the script.

Comment (tatiana) [reply tests/benchmark/run.py:32]: @sunank200 @utkarsharma2 We're adapting the benchmark to run as K8s jobs, as part of #432  so I believe the `dag_id` won't be an issue (each job will be run in a separate pod) - WDYT?

**Review by tatiana — APPROVED** [review]
LGTM, thanks, @utkarsharma2 !

**Pull request merged**